src/ChronosTime.php

```diff
@@ -143,11 +188,11 @@ public function getSeconds(): int
      */
     public function setSeconds(int $seconds): static
     {
-        $prevSecondsTicks = $this->ticks % self::TICKS_PER_MINUTE - $this->ticks % self::TICKS_PER_SECOND;
-        $newSecondsTicks = static::mod($seconds, 60) * self::TICKS_PER_SECOND;
+        $baseTicks = $this->ticks - ($this->ticks % self::TICKS_PER_MINUTE - $this->ticks % self::TICKS_PER_SECOND);
+        $newTicks = static::mod($baseTicks + $seconds * self::TICKS_PER_SECOND, self::TICKS_PER_DAY);
```

Comment:
This now adds all the rollover to ticks before performing the mod() on ticks per day.